### PR TITLE
[4.0] Implement Field group buttons as Custom Element

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes

The radio field has few dedicated layouts/functionalities:
- native radios
- group buttons (ie bootstrap group buttons, used extremely in J3)
- switcher (the default on J4)

So this PR removes the code that used to be in the templates file template.js to control those fields (yeah very strange) and introduces a custom element `joomla-field-group-buttons` that provides the same functionality. Also since it's CE it works as is with the repeatable forms, no more custom events, etc

### Testing Instructions
For testing purposes you can find couple switches in the admin->extensions->templates->autum tab advanced. You can play around, save etc. Everything should work fine

This code in the xml will be removed prior to (if ever) merging this PR   

### Actual result

screen shot 2018-05-23 at 14 26 34

Also plain html/js/css demo: https://codepen.io/dgrammatiko/pen/LmvaNa

### Documentation Changes Required
Not really, we are still B/C. Only consideration here is for devs that using those form fields out of JForm. In such case they need to swap the div tag with the new joomla-field-group-button and also include the corresponding js for the custom element, similar to what is done in the layout of the field.

### CODE REVIEW

**Comment by dgrammatiko** [comment]
@ciar4n can you tide up the css here?

**Comment by ciar4n** [comment]
Now looking like this (inline with new design)...

image

File: build/webcomponents/scss/field-group-buttons/field-group-buttons.scss [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    &.btn-primary,
+    &.btn-success,
+    &.btn-info,
+    &.btn-warning,
+    &.btn-danger {
+      color: inherit;
+      background-color: #e6e6e6;
+    }
+
+
```
Comment (Quy) [inline build/webcomponents/scss/field-group-buttons/field-group-buttons.scss]: Remove blank line.

**Comment by brianteeman** [comment]
@dgrammatiko can you resolve the conflicts please so this can be tested

**Comment by brianteeman** [comment]
Same problem as with the switcher. The label of the radio button is never announced to the screenreader

**Comment by dgrammatiko** [comment]
@brianteeman this is in essence an exact copy of the BS markup: 
https://getbootstrap.com/docs/4.1/components/buttons/#checkbox-and-radio-buttons

But if you have a solution here I'm happy to code that in

**Comment by brianteeman** [comment]
those examples work correctly with a screenreader

**Comment by dgrammatiko** [comment]
The idea and the aim for J4 was to get **all** fields as custom elements as those are lazily loaded and could actually run ES20XX code, but it seems that this has changed lately. So let it be. Closing

**Pull request closed without merging**